server/services/task-manager.js

```diff
@@ -0,0 +1,80 @@
+const pokemonClient = require("../clients/pokemon-client.js");
+const file = require("./file-manager.js");
+
+class TaskManager {
+   getTasks = () => file.getAllTasks();
+
+   async add(task) {
+      const allPokemonNames = await file.getFilePokemonNames();
+      if (
+         task
+            .replace(/\s/g, "")
+            .split(",")
+            .every((elem) => !isNaN(elem) || allPokemonNames.includes(elem))
+      ) {
+         let pokemonIDS = task.replace(/\s/g, "").split(","); // "1, 2, 3" => [1,2,3]
+         const pokemonData = await Promise.all(
+            pokemonIDS.map((id) => pokemonClient.getPokemon(id))
+         );
+         pokemonData.forEach((pokemon, i) => {
+            this._pokemonTasksHandle(pokemon, pokemonIDS, i);
+         });
+      } else {
+         file.addTask(new Task(task));
+      }
+   }
+
+   remove = (i) => file.deleteTask(i);
+
+   clear = () => file.clearTasks();
+
+   sort(direction) {
+      let sortedTasks;
+      sortedTasks = this.getTasks().sort((a, b) => {
+         if (a.taskText < b.taskText) return 1;
+         return -1;
+      });
+      if (direction === "down") {
+         sortedTasks = sortedTasks.reverse();
+      }
+      file.clearTasks();
+      sortedTasks.forEach((task) => file.addTask(task));
+   }
+
+   _pokemonTasksHandle(pokemon, pokemonIDS, i) {
```

Comment:
`i` is not a good variable name. maybe `index`?
anyways, in this case you should have just received the pokemon ID instead - making your method interface much more simple
```suggestion
   _pokemonTasksHandle(pokemon, pokemonID) {
```

Reply:
Wow thanks, I really didnt need to pass index and just pass pokemonIDS[index]